Solutions/CiscoWSA/Parsers/CiscoWSAEvent.txt

```diff
@@ -0,0 +1,168 @@
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+// **** Access Logs (non-w3c format) ****
+// This Kusto function expects the following fiels order for CiscoWSA Access logs (Squid type):
+// %t %e %a %w %h %s %1r %2r %A %H %d %c %D %Xr %?BLOCK_SUSPECT_ USER_AGENT, MONITOR_SUSPECT_ USER_AGENT?% < User-Agent:%!%-%
+// **** Access Logs (w3c format) ****
+// This Kusto function expects the following fiels order for CiscoWSA logs in w3c format:
+// date time timestamp x-elapsed-time referer c-ip sc-result-code cs-bytes sc-body-size cs-method cs-url s-hostname cs-uri cs-username "cs-mime-type" x-acltag X-Forwarded-For c-port s-computerName s-port cs-version x-webcat-code-abbr x-wbrs-score x-webroot-scanverdict x-webroot-threat-name x-webroot-trr x-webroot-spyid x-webroot-trace-id x-mcafee-scanverdict x-mcafee-filename x-mcafee-av-scanerror x-mcafee-av-detecttype x-mcafee-av-virustype x-mcafee-virus-name x-sophos-scanverdict x-sophos-scanerror x-sophos-file-name x-sophos-virus-name x-ids-verdict x-icap-verdict x-webcat-req-code-abbr x-webcat-resp-code-abbr x-wbrs-threat-type x-avc-app x-avc-type x-avc-behavior x-request-rewrite x-avg-bw x-bw-throttled user-type x-resp-dvs-threat-name x-resp-dvs-scanverdict x-resp-dvs-verdictname x-req-dvs-threat-name x-req-dvs-scanverdict x-req-dvs-verdictname x-amp-verdict x-amp-malware-name x-amp-score x-amp-upload x-amp-filename x-amp-sha x-suspect-user-agent bytes
+// Description of log fields for different formats can be found here: https://www.cisco.com/c/en/us/td/docs/security/wsa/wsa_14-0/User-Guide/b_WSA_UserGuide_14_0/b_WSA_UserGuide_11_7_chapter_010101.html#con_1621159
+let cisco_wsa_access_logs =() {
+Syslog
+| where SyslogMessage matches regex @"\A\d{10}\.\d{3}\s\d+\s\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}"
+| extend EventVendor = 'Cisco'
+| extend EventProduct = 'Web Security Appliance'
+| extend EventType = 'Access Log'
+| extend EventFields = split(SyslogMessage, ' ')
+| extend ScanningVerdictFields = parse_csv(tostring(extract(@'<(.*?)>', 1, SyslogMessage)))
+| extend EventStartTime = unixtime_seconds_todatetime(todouble(EventFields[0]))
+| extend Latency = toint(EventFields[1])
+| extend SrcIpAddr = tostring(EventFields[2])
+| extend EventResultDetails = extract(@'\A(.*?)\/[1-5]\d{2}', 1, tostring(EventFields[3]))
+| extend HttpStatusCode = extract(@'\A.*?\/([1-5]\d{2})', 1, tostring(EventFields[3]))
+| extend DstBytes = toint(EventFields[4])
+| extend HttpRequestMethod = tostring(EventFields[5])
+| extend UrlOriginal = tostring(EventFields[6])
+| extend SrcUserName = tostring(EventFields[7])
+| extend ContactedServerCode = extract(@'\A(\w+)\/\d{1,3}', 1, tostring(EventFields[8]))
+| extend DstIpAddr = extract(@'\A\w+\/(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})', 1, tostring(EventFields[8]))
+| extend DstHostname = extract(@'\A\w+\/(\D+)', 1, tostring(EventFields[8]))

```

Comment:
It should not DstDvcHostname.Please refer the guideline [here](https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema).

Thanks,
Aditi Maheshwari

Reply:
updated